time_stamper_cv/include/configuration.h

```diff
@@ -41,10 +41,28 @@ struct LedRowConfig {
 typedef std::map<std::string, LedRowConfig> LedRowConfigRepository;
 
 struct ImageProcessorConfig {
+  inline static constexpr const char *TOP_ROW{"TopRow"};
+  inline static constexpr const char *BOTTOM_ROW{"BottomRow"};
+
   int tolerance{10};
   cv::SimpleBlobDetector::Params params;
   LedRowConfigRepository led_row_config {
-      {"BottomRow", {{0, 0}, {6, 0}, 16}},
-      {"TopRow", {{0, 17}, {6, 0}, 10}}
+      {BOTTOM_ROW, {{0, 0}, {6, 0}, 16}},
+      {TOP_ROW, {{0, 17}, {6, 0}, 10}}
   };
+
+  /**
+   * Get row names
+   * @return string vector
+   */
+ public:
+  std::vector<std::string> rows() const {
+    std::vector<std::string> res{};
+
+    std::map<std::string, LedRowConfig>::const_iterator it;
+    for (it = led_row_config.begin() ; it != led_row_config.end(); it++) {
+      res.push_back(it->first);
+    }
+    return res;
```

Comment:
This you can greatly simplify, depending on your version of c++. I'd assume at least 11. Otherwise something is quite off here.
 
```suggestion
    std::vector<std::string> res{};
    for (const auto &item : led_row) {
      res.push_back(item.first);
    }
    return ret;
```
With c++17 you can destructure the key-value pair (`item`) to make it nicer, `for(const auto& [key, _] : led_rows) ...`.

With c++20 you could initialize from a view instead:
```suggestion
    const auto kv = std::views::keys(led_rows);
    return { kv.begin(), kv.end() };
```
c++23 will probably further simplify :) (`.to`). At least they are getting there :D.

Generally this pattern has a quite sub-optimal performance because it creates the same vector on each call. Not a sign. problem here, but you should know. 
To address that I'd use a static variable and IIFE to initialize it. Here a c++20 compatible example but you can do even with c++11.

```suggestion
    const static auto row_names = []() {
       const auto kv = std::views::keys(led_rows);
       return std::vector<std::string>{ kv.begin(), kv.end() };
    }();
    return row_names;
```
(IMPORTANT: this makes only sense if you change the return type to a const ref at the same time)

Another performance sub-optimality is the needless copy of a std::string. A vector of `string_view` s could make that better since c++17. 
But no need to do that -- optimization is important to have a good basic feeling for but should usually be done with care -- if it make the code more complicated because that is usually the more problematic cost foremost because it leads to bugs!